datablock.py

```diff
@@ -119,26 +119,18 @@ def unique_beams(self):
             else:
                 for i in range(len(iset)):
                     obj[iset.get_beam(i)] = None
-        return list(obj.keys())
+        return list(obj)
 
     def unique_detectors(self):
-        """ Returns a list of detector objects. """
-        return list(self._unique_detectors_dict().keys())
-
-    def _unique_detectors_dict(self):
-        """ Returns an ordered dictionary of detector objects. """
+        """ Returns an ordered list of detector objects. """
```

Comment:
Method name is `unique_detectors_dict` yet returns a `list`

Though Looking at this in isolation I cannot fathom how a dictionary keyed by detectors with values of detector number was _useful_?